src/source/Utils/ExponentialBackoffUtils.h

```diff
@@ -0,0 +1,139 @@
+/******************************************************
+A generic utility for exponential back off waits
+******************************************************/
+#ifndef __KINESIS_VIDEO_EXPONENTIAL_BACKOFF_HELPER_INCLUDE__
+#define __KINESIS_VIDEO_EXPONENTIAL_BACKOFF_HELPER_INCLUDE__
+
+#pragma once
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <com/amazonaws/kinesis/video/webrtcclient/Include.h>
+
+/************************************************************************
+ Following #define values are default configuration values for
+ exponential backoff retries.
+
+ With these values, the wait times will look like following -
+    ************************************
+    * Retry Count *      Wait time     *
+    * **********************************
+    *     1       *    300ms + jitter  *
+    *     2       *    600ms + jitter  *
+    *     3       *   1200ms + jitter  *
+    *     4       *   2400ms + jitter  *
+    *     5       *   4800ms + jitter  *
+    *     6       *   9600ms + jitter  *
+    *     7       *  10000ms + jitter  *
+    *     8       *  10000ms + jitter  *
+    *     9       *  10000ms + jitter  *
+    ************************************
+ jitter = random number between [0, 300)ms.
+************************************************************************/
+/**
+ * Factor for computing the exponential backoff wait time
+ */
+#define DEFAULT_RETRY_TIME_FACTOR_MILLISECONDS                  300
+
+/**
+ * Factor determining the curve of exponential wait time
+ */
+#define DEFAULT_EXPONENTIAL_FACTOR                              2
```

Comment:
Consider adding specificity to the naming to avoid any potential collisions. While the unit tests and the sample might not have collisions, the client applications might be pulling in some libraries which might collide

Reply:
Makes sense. Changing all #defines from DEFAULT_* to DEFAULT_KVS_*
Could not think of a better name. But this will most likely not collide.